docs/reference/remap2/literals/regular_expression.cue

```diff
@@ -0,0 +1,58 @@
+package metadata
+
+remap2: literals: regular_expression: {
+	title:       "Regular Expression"
+	description: """
+		A Regular Expression literal represents a [Regular Expression](\(urls.regex)) used for string matching and
+		parsing.
+
+		Regular expressions are delimited by `/` and use [Rust regex syntax](\(urls.rust_regex_syntax)). There are a
+		few limitations with Regular Expressions in VRL:
+
+		1. You can't assign a regex to an event path. Thus, `.pattern = /foo|bar/i` is not
+		   allowed.
+		2. Expressions can't return regexes. Thus you can't, for example, dynamically create
+		   regular expressions.
```

Comment:
The second part is true, the first one not exactly because `foo = /bar/` does return the regular expression as the result of the assignment expression.